### ISSUE TITLE
New version "2.1.2-nightly.347274.c2cbc47" not working with existing project

### ISSUE THREAD

**Issue by Levis0045** [microsoft/BotFramework-Composer#9557]
## Describe the bug

I have some projects with Bot Framework Composer and they are working fine on version 2.1.2, with auto ugrade enabled.

Since 11/04/2024 after the auto installation of version "2.1.2-nightly.347274.c2cbc47", all my projects are not working again.

## Version

![image](https://user-images.githubusercontent.com/13846122/231430029-f558d3ee-b1dc-409a-8735-d677d03fb7e6.png)

## Browser

- [X] Electron distribution
- [ ] Chrome
- [ ] Safari
- [ ] Firefox
- [ ] Edge

## OS

- [ ] macOS
- [ ] Windows
- [X] Ubuntu

## To Reproduce

1. Open existing project
2. Start the bot (by compiling it)
3. Get some errors on console (Web Chat pannel)
4. Not response on test chat channel

## Expected behavior

Bot Composer project should work between versions 

## Screenshots

![image](https://user-images.githubusercontent.com/13846122/231430795-f3f85694-b99e-41dd-b25c-7ed3c41cba71.png)

![image](https://user-images.githubusercontent.com/13846122/231431492-36708522-5b35-495d-8092-64ec8ff57090.png)

**Comment by OEvgeny** [comment]
@Levis0045 does the issue happen with a newly created bots? Does it affect NodeJS bots or only .Net? Could you provide the developer tools console logs for the issue?

Suggesting downgrading the Composer manually for now

**Comment by Levis0045** [comment]
Hi @OEvgeny,

It only happen on project build with version **2.1.2** with **.NET**. I have Downgraded to the last version. 

I received new update yesterday (another version of **2.1.2** in remplement of **2.1.2-nightly.347274.c2cbc47** version ). 

Another problem happen with this new update. After starting and selecting bot project, Bot Composer send continiuos POST requests with response 500 (1000 par second). Please check console log:

```json
{
  "error": {},
  "id": "3d5328d2-4dc9-4cdb-95c1-a68929498aed",
  "request": {
    "method": "POST",
    "payload": {
      "type": "trace",
      "serviceUrl": "http://localhost:5000",
      "channelId": "emulator",
      "from": {
        "id": "d169c709-15fb-4db3-b913-5c23669f19e7",
        "name": "Bot",
        "role": "bot"
      },
      "conversation": {
        "id": "e31de584-237f-448b-a91f-6c6c47538b83|livechat"
      },
      "recipient": {
        "id": "1649e6f9-4779-4137-bb90-27171d8c7719",
        "name": "",
        "role": "user"
      },
      "locale": "fr-fr",
      "replyToId": "90ad2766-53c9-4d97-89f4-2053d9b2c552",
      "label": "Bot State",
      "valueType": "https://www.botframework.com/schemas/botState",
      "value": {
        "turn": {
          "locale": "fr-fr",
          "activity": {
            "type": "message",
            "id": "90ad2766-53c9-4d97-89f4-2053d9b2c552",
            "timestamp": "2023-04-12T11:56:33.976Z",
            "localTimestamp": "2023-04-12T11:56:33Z",
            "localTimezone": "Europe/Paris",
            "serviceUrl": "http://localhost:5000",
            "channelId": "emulator",
            "from": {
              "id": "1649e6f9-4779-4137-bb90-27171d8c7719",
              "name": "",
              "role": "user"
            },
            "conversation": {
              "id": "e31de584-237f-448b-a91f-6c6c47538b83|livechat"
            },
            "recipient": {
              "id": "d169c709-15fb-4db3-b913-5c23669f19e7",
              "name": "Bot",
              "role": "bot"
            },
            "textFormat": "plain",
            "locale": "fr-fr",
            "text": "cc",
            "attachments": [],
            "entities": [
              {
                "type": "ClientCapabilities",
                "requiresBotState": true,
                "supportsListening": true,
                "supportsTts": true
              }
            ],
            "channelData": {
              "clientActivityID": "1681300593932c6rullfvt3"
            },
            "callerId": "urn:botframework:azure"
          },
          "dialogEvent": {
            "Bubble": false,
            "Name": "endOfActions",
            "Value": null
          },
          "recognized": {
            "text": "cc",
            "alteredText": null,
            "intents": {
              "None": {
                "score": 1
              }
            },
            "entities": {
              "$instance": {}
            },
            "intent": "None",
            "score": 1
          },
          "interrupted": true,
          "lastresult": null
        },
        "dialog": {
          "eventCounter": 1337995,
          "_adaptive": {
            "actions": []
          },
          "lastIntent": "None",
          "lastEvent": "endOfActions"
        },
        "this": {
          "eventCounter": 1337995,
          "_adaptive": {
            "actions": []
          },
          "lastIntent": "None",
          "lastEvent": "endOfActions"
        },
        "conversation": {
          "DialogState": {
            "dialogStack": [
              {
                "id": "CoreWithQnA",
                "state": {
                  "eventCounter": 1337995,
                  "_adaptive": {
                    "actions": []
                  },
                  "lastIntent": "None",
                  "lastEvent": "endOfActions"
                },
                "StackIndex": null,
                "Version": "OPMnpbVwjGnKlZ41OUoqyFYQho11+L9yPmPe2qorUWM="
              }
            ]
          }
        },
        "user": {}
      },
      "name": "BotState"
    },
    "route": "/v3/conversations/e31de584-237f-448b-a91f-6c6c47538b83%7Clivechat/activities/90ad2766-53c9-4d97-89f4-2053d9b2c552"
  },
  "response": {
    "payload": {
      "message": "Conversation not found. BadArgument"
    },
    "statusCode": 500
  },
  "timestamp": 1681378138978
}
```
After 5 min, i received this error and program crash : 

![image](https://user-images.githubusercontent.com/13846122/231719577-8e354c3e-a785-4141-8c65-8b349cefa1a1.png)

Thanks

**Comment by OEvgeny** [comment]
Hi @Levis0045,

Thank you for the additional context.

Does the issue from the title still happen in the latest build for you?

**Comment by Levis0045** [comment]
Hi @OEvgeny,

Yes is still happen on the latest build.

**Comment by OEvgeny** [comment]
I'm still unable to reproduce the main issue on my end. New nightly properly closes Composer which might be the cause.

@Levis0045 does the main issue happen only with bots with QnA set up? There must be something specific to the bot not stated in the issue.

**Comment by Levis0045** [comment]
Hi @OEvgeny,

It happen bot on QnA set up and when we us Question answering REST API request. I don't know why Bot Composer send loop request.

**Comment by ramfattah** [comment]
Hi  @Levis0045, I'm investigating this issue.

**Comment by ramfattah** [comment]
Hi @Levis0045, 

I'm not able to reproduce this issue with a fresh CoreWithQnA Bot Template that was created in Composer 2.1.2 and tested with the latest Composer nightly build (currently [39f01c2](https://github.com/microsoft/BotFramework-Composer-Nightlies/releases/tag/v2.1.2-nightly.348135.39f01c2)).

@Levis0045, are you able to provide a minimal reproducible .zip composer project that reproduces this issue?

**Comment by Levis0045** [comment]
Hi @ramfattah 

Thanks for your investigation. Eventually I found a solution to the problem. But, this problem deserves to be treated.

Turns out it had a skill server endpoint port misconfiguration (skillHostEndpoint=http://localhost:3980/api/skills instead of skillHostEndpoint=http://localhost:3984/api/ skills). Bot composer was trying to send requests to this API on the wrong port without success. And instead of stopping on error, it kept sending endless requests with 500 responses until it crashed Bot Composer. This appear in the two last versions.

I also found that the error was due to the Bot Composer cache. The more the application is used (by restarting the test server to apply the new modifications), the more Bot Composer becomes very slow and less performance in the final rendering. The solution I found is to completely stop my machine and then turn it on. This shutdown makes Bot Composer operational again for a while.

I think we should also improve the compilation system and the error management system to better guide developers or users in their development practices.
